docs/source/en/tasks/knowledge_distillation_for_image_classification.md

```diff
@@ -0,0 +1,188 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+# Knowledge Distillation for Computer Vision
+
+Knowledge distillation is a technique used to transfer knowledge from a larger, more complex model (teacher) to a smaller, simpler model (student). In the context of image classification, the goal is to train the student model to mimic the behavior of the teacher model. In this notebook, we will do task-specific knowledge distillation. We will use [beans](https://huggingface.co/datasets/beans) for this.
+
+This tutorial aims to demonstrate how to distil ResNet (teacher model)[https://huggingface.co/microsoft/resnet-50] to [MobileNet](https://huggingface.co/google/mobilenet_v2_1.4_224) (student model) using `Trainer` API of 🤗 Transformers.
+
+Let's install the libraries needed for distillation and evaluating the process. 
+
+```bash
+pip install transformers datasets accelerate tensorboard evaluate --upgrade
+```
+
+In this example, we are using `google/mobilenet_v2_1.4_224` and `microsoft/resnet-50`, both are trained with ImageNet-1k dataset on resolution of 224x224. You can see that feature extractor returns the same output below.
+
+```python
+from transformers import AutoFeatureExtractor
+from PIL import Image
+import requests
+import numpy as np
+
+teacher_extractor = AutoFeatureExtractor.from_pretrained("microsoft/resnet-50")
+student_extractor = AutoFeatureExtractor.from_pretrained("google/mobilenet_v2_1.4_224")
+
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+sample = Image.open(requests.get(url, stream=True).raw)
+
+np.array_equal(teacher_extractor(sample),student_extractor(sample))
+```
+
+We will now load and pre-process the dataset. 
+
+```python
+from datasets import load_dataset
+
+dataset = load_dataset("beans")
+```
+
+We can use either of the processors given they return the same output. We will use `map()` method of `dataset` to apply the preprocessing to every split of the dataset.
```

Comment:
> We can use either of the processors given they return the same output.

Do they use the exact same normalization of pixels? They actually don't. Looking at the preprocessor configs ([here](https://huggingface.co/microsoft/resnet-50/blob/main/preprocessor_config.json) and [here](https://huggingface.co/google/mobilenet_v2_1.4_224/blob/main/preprocessor_config.json)), we can see that their `image_mean` and `image_std` stats are different. So, we need to be careful here. 

If we're randomly initializing the student model (which I think we should for distillation), then we need to either calculate the `image_mean` and `image_std` stats from the underlying dataset (`beans` in this case). 

Or, we could just scale the pixel values to be either in the [0, 1] or [-1, 1] range. 

Also, the teacher model needs to be fine-tuned on the `beans` dataset here. We shouldn't be using an ImageNet-1k pre-trained model here.